tensorflow_quantum/core/qsim/fuser_basic.cc

```diff
@@ -31,17 +31,75 @@ using ::tensorflow::Status;
 // Appends to `fused_gates` all single-qubit gates on the current `qubit_wire`,
 // until reaching the end of the wire or a multi-qubit gate.
 // Starts the search at `current_timeslice`.
-void Advance(const std::vector<const Gate*>& qubit_wire,
-             std::vector<const Gate*>* fused_gates,
+void Advance(const std::vector<const Gate*>& qubit_wire, GateFused* gate,
              unsigned int* current_timeslice) {
   while (*current_timeslice < qubit_wire.size() &&
          qubit_wire[*current_timeslice]->num_qubits == 1) {
-    fused_gates->push_back(qubit_wire[(*current_timeslice)++]);
+    gate->AddGate(qubit_wire[(*current_timeslice)++]);
   }
 }
 
 }  // namespace
 
+GateFused::GateFused(const unsigned int time, const unsigned int q0,
```

Comment:
should it be `const Gate& anchor` instead ? it looks like you are passing in `&a_gate` in a lot of the tests

Reply:
`&a_gate` passes in a pointer, while `const Gate& anchor` expects a reference, right?  I don't think a reference would work here because we are storing a way to look at the anchor gate again later, rather than mutating an external object